### PR TITLE
docs: update changelog for 1.21.0

### PR DESCRIPTION
**Pull request by KazuCocoa**
## Proposed changes

Xcode 12.5 is coming, so it's a good time to bump the current RC to 1.21.0

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,48 @@
+CHANGES IN VERSION 1.21.0 (FROM 1.20.2)
+===================================
+
+Appium 1.21.0 is a minor release
+
+### General
+
+* Minimum NodeJS version moved to 12
+
+### Android General
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:10]: Please mention 
- the recent adb fixes for `mobile: type` command:  https://github.com/appium/appium-adb/pull/567
- addition of speed parameter to mock location calls: https://github.com/appium/appium-adb/pull/563

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,48 @@
...
+
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:11]: Switching to web views with same package name but different process names has been fixed: https://github.com/appium/appium-android-driver/pull/687

Comment (mykola-mokhnach) [reply CHANGELOG.md:11]: Sending of Google analytics data by instruments has been disabled

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,48 @@
...
+### Android(UIAutomator2)
+
+### Android(Espresso)
+
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:15]: Gradle wrapper has been rewritten using Kotlin DSL: https://github.com/appium/appium-espresso-driver/pull/659

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,48 @@
...
+* Xcode 12.5 support
+
+### iOS(XCUITest)
+
+* Add capabilities:
+    * `appium:includeDeviceCapsToSessionInfo` to skip adding the screen information in order to improve [Get Session Capabilities](http://appium.io/docs/en/commands/session/get/) API. `mobile:viewportRect` and `mobile:deviceScreenInfo` help to get these information when this capability is `false`. [appium-xcuitest-driver#1278](https://github.com/appium/appium-xcuitest-driver/pull/1278)
+    * `appium:disableAutomaticScreenshots` to handle automatic screenshot by XCTest. Defaults to `true` [appium-xcuitest-driver#1288](https://github.com/appium/appium-xcuitest-driver/pull/1288)
+    * `appium:shouldTerminateApp` to handle if Appium terminates the application under test process on the session end. Defaults to `true` [appium-xcuitest-driver#1290](https://github.com/appium/appium-xcuitest-driver/pull/1290), [WebDriverAgent#486/](https://github.com/appium/WebDriverAgent/pull/486/)
+* Add `mobile:` functions:
+    * `mobile:viewportRect` returns the screen viewport information which is part of [Get Session Capabilities](http://appium.io/docs/en/commands/session/get/)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:27]: this one was already there for a while

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,48 @@
...
+    * `mobile:performIoHidEvent` to emulate low-level IO HID device event. Read [mobile:performIoHidEvent](https://github.com/appium/appium-xcuitest-driver#mobile-performiohidevent) for more details. [appium-xcuitest-driver#1291](https://github.com/appium/appium-xcuitest-driver/pull/1291), [WebDriverAgent#494](https://github.com/appium/WebDriverAgent/pull/494)
+* Support css selector [appium-xcuitest-driver#1274](https://github.com/appium/appium-xcuitest-driver/pull/1274)
+* Support [Get Geolocation](https://appium.io/docs/en/commands/session/geolocation/get-geolocation/) API [appium-xcuitest-driver#1266](https://github.com/appium/appium-xcuitest-driver/pull/1266)
+    * _Allow Location Access_ in _Location Services_ for _WebDriverAgent-Runner_ must be **Always**
+* Add `isAccessible` field in the page source [WebDriverAgent#493](https://github.com/appium/WebDriverAgent/pull/493)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:35]: the field name in the XML is `accessible`

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,48 @@
...
+* Improve performance of the keyboard dismissal logic [appium-xcuitest-driver#1267](https://github.com/appium/appium-xcuitest-driver/pull/1267), [WebDriverAgent#458](https://github.com/appium/WebDriverAgent/pull/458)
+* Fix performance degration of interaction in `nativeWebTap: true` [appium-xcuitest-driver/pull/1265](https://github.com/appium/appium-xcuitest-driver#1265)
+
+### iOS(Safari)
+
+### Mac2
+
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:42]: - More fine-grained handling of orphaned xcodebuild processes: https://github.com/appium/appium-mac2-driver/pull/73
- Added `hittable` attribute to the page source: https://github.com/appium/appium-mac2-driver/pull/65
- Added a possibility to provide key names for `macos: key` extension: https://github.com/appium/appium-mac2-driver/pull/60

File: CHANGELOG.md (hunk 1 above)
Comment (KazuCocoa) [reply CHANGELOG.md:11]: note: added `mobile:type`. The speed is not yet.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,59 @@
...
+### General
+
+* Minimum NodeJS version moved to 12
+* Support multiple image matches for [Finding and Interacting with Image Elements](http://appium.io/docs/en/advanced-concepts/image-elements/) [appium-base-driver#449](https://github.com/appium/appium-base-driver/pull/449)
+
+### Android General
+
+* Fix `autoLaunch` to install applications by `app` and `otherApps` capabilities, but does not launch them as same as iOS [appium-uiautomator2-driver#440](https://github.com/appium/appium-uiautomator2-driver/pull/440), [appium-espresso-driver#654](https://github.com/appium/appium-espresso-driver/pull/654), [appium-android-driver#676](https://github.com/appium/appium-android-driver/pull/676)
+* Fix to handle special charactors by `mobile:type` [appium-adb#567](https://github.com/appium/appium-adb/pull/567)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:14]: characters

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,59 @@
...
+* Disable to send of Google analytics data by instruments [https://github.com/appium/appium-uiautomator2-driver/pull/452](appium-uiautomator2-driver#452), [appium-espresso-driver#666](https://github.com/appium/appium-espresso-driver/pull/666)
+
+### Android(UIAutomator2)
+
+* Add `mobile:` functions:
+    * `mobile:doubleClickGesture` for double click guesture [appium-uiautomator2-driver#441](https://github.com/appium/appium-uiautomator2-driver/pull/441), [appium-uiautomator2-server#415](https://github.com/appium/appium-uiautomator2-server/pull/415)
+    * `mobile:installMultipleApks` to install multiple apks via `adb install-multiple` command. Please read [mobile:installMultipleApks](https://github.com/appium/appium-uiautomator2-driver/blob/master/README.md#mobile-installmultipleapks) for more details
+* Fix XPath lookup for queries which has `@*` attribute pattern [appium-uiautomator2-server#417](https://github.com/appium/appium-uiautomator2-server/pull/417)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:22]: which have

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,59 @@
...
+### General
+
+* Minimum NodeJS version moved to 12
+* Support multiple image matches for [Finding and Interacting with Image Elements](http://appium.io/docs/en/advanced-concepts/image-elements/) [appium-base-driver#449](https://github.com/appium/appium-base-driver/pull/449)
+
+### Android General
+
+* Fix `autoLaunch` to install applications by `app` and `otherApps` capabilities, but does not launch them as same as iOS [appium-uiautomator2-driver#440](https://github.com/appium/appium-uiautomator2-driver/pull/440), [appium-espresso-driver#654](https://github.com/appium/appium-espresso-driver/pull/654), [appium-android-driver#676](https://github.com/appium/appium-android-driver/pull/676)
+* Fix to handle special charactors by `mobile:type` [appium-adb#567](https://github.com/appium/appium-adb/pull/567)
+* Disable to send of Google analytics data by instruments [https://github.com/appium/appium-uiautomator2-driver/pull/452](appium-uiautomator2-driver#452), [appium-espresso-driver#666](https://github.com/appium/appium-espresso-driver/pull/666)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:15]: Disable sending

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,67 @@
...
+Appium 1.21.0 is a minor release
+
+### General
+
+* Minimum NodeJS version moved to 12
+* Support multiple image matches for [Finding and Interacting with Image Elements](http://appium.io/docs/en/advanced-concepts/image-elements/) [appium-base-driver#449](https://github.com/appium/appium-base-driver/pull/449)
+
+### Android General
+
+* Disable to sending of Google analytics data by instruments [https://github.com/appium/appium-uiautomator2-driver/pull/452](appium-uiautomator2-driver#452), [appium-espresso-driver#666](https://github.com/appium/appium-espresso-driver/pull/666)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:13]: Disable sending of ...

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,67 @@
...
+* Improve to detect screen lock status [appium/appium-adb#564](https://github.com/appium/appium-adb/pull/564)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:14]: Improve detection of ...

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,67 @@
...
+* Add `speed` poptional arameter to mock location [appium-android-driver#681](https://github.com/appium/appium-android-driver/pull/681)
+* Fix `autoLaunch` to install applications by `app` and `otherApps` capabilities, but does not launch them as same as iOS [appium-uiautomator2-driver#440](https://github.com/appium/appium-uiautomator2-driver/pull/440), [appium-espresso-driver#654](https://github.com/appium/appium-espresso-driver/pull/654), [appium-android-driver#676](https://github.com/appium/appium-android-driver/pull/676)
+* Fix to handle special characters by `mobile:type` [appium-adb#567](https://github.com/appium/appium-adb/pull/567)
+* Fix to switch to Web Views with same package name, but different process names [appium-android-driver#687](https://github.com/appium/appium-android-driver/pull/687)
+
+### Android(UIAutomator2)
+
+* Add `mobile:` functions:
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:22]: functions  -> extensions

Comment (mykola-mokhnach) [reply CHANGELOG.md:22]: same below

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,67 @@
...
+* Fix XPath lookup for queries which have `@*` attribute pattern [appium-uiautomator2-server#417](https://github.com/appium/appium-uiautomator2-server/pull/417)
+* Fix to stop screen recording after a session is closed [appium-uiautomator2-driver#439](https://github.com/appium/appium-uiautomator2-driver/pull/439)
+
+### Android(Espresso)
+
+* Fix to raise an error in [Reset App](http://appium.io/docs/en/commands/device/app/reset-app/). It must quit the session and create a new one in Espresso driver. [appium-espresso-driver#649](https://github.com/appium/appium-espresso-driver/pull/649)
+* Fix to reset instrumentation state before starting a new instance [appium-espresso-driver#662](https://github.com/appium/appium-espresso-driver/pull/662)
+* Gradle wrapper has been rewritten using Kotlin DSL [appium-espresso-driver#658](https://github.com/appium/appium-espresso-driver/pull/658), [appium-espresso-driver#659](https://github.com/appium/appium-espresso-driver/pull/659)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:32]: I would say it makes sense to explicitly mention people who don't belong to the team and who dedicated their time to improve Appium here

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,67 @@
...
+    * `appium:shouldTerminateApp` to handle if Appium terminates the application under test process on the session end. Defaults to `true` [appium-xcuitest-driver#1290](https://github.com/appium/appium-xcuitest-driver/pull/1290), [WebDriverAgent#486/](https://github.com/appium/WebDriverAgent/pull/486/)
+* Add `mobile:` functions:
+    * `mobile:viewportRect` returns the screen viewport information which is part of [Get Session Capabilities](http://appium.io/docs/en/commands/session/get/)
+    * `mobile:deviceScreenInfo` returns the screen information which is part of [Get Session Capabilities](http://appium.io/docs/en/commands/session/get/), [WebDriverAgent#483](https://github.com/appium/WebDriverAgent/pull/483)
+    * `mobile:pushNotification` to simulator push notification on simulator by `xcrun simctl push` command. It requires Xcode 11.4+. Please read [mobile:pushNotification](https://github.com/appium/appium-xcuitest-driver#mobile-pushnotification) for mroe details. [appium-xcuitest-driver#1286](https://github.com/appium/appium-xcuitest-driver/pull/1286/)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:48]: more details, to ~~simulator~~ trigger a push notification

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,67 @@
...
+    * `mobile:expectNotification` to wait for a notification on the device under test. Please read [mobile:expectNotification](https://github.com/appium/appium-xcuitest-driver#mobile-pushnotification) to see how to use it. [appium-xcuitest-driver#1287](https://github.com/appium/appium-xcuitest-driver/pull/1287) [WebDriverAgent#480](https://github.com/appium/WebDriverAgent/pull/480)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:49]: ... for more information on how to use it

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,67 @@
...
+    * `mobile:performIoHidEvent` to emulate low-level IO HID device event. Read [mobile:performIoHidEvent](https://github.com/appium/appium-xcuitest-driver#mobile-performiohidevent) for more details. [appium-xcuitest-driver#1291](https://github.com/appium/appium-xcuitest-driver/pull/1291), [WebDriverAgent#494](https://github.com/appium/WebDriverAgent/pull/494)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:50]: device events

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,67 @@
...
+* Support css selector [appium-xcuitest-driver#1274](https://github.com/appium/appium-xcuitest-driver/pull/1274)
+* Support [Get Geolocation](https://appium.io/docs/en/commands/session/geolocation/get-geolocation/) API [appium-xcuitest-driver#1266](https://github.com/appium/appium-xcuitest-driver/pull/1266)
+    * _Allow Location Access_ in _Location Services_ for _WebDriverAgent-Runner_ must be **Always**
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:53]: must be set to ...

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,67 @@
...
+* Add `accessible` field in the XML page source [WebDriverAgent#493](https://github.com/appium/WebDriverAgent/pull/493)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:54]: into the XML

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,3 +1,67 @@
...
+* Improve performance of the keyboard dismissal logic [appium-xcuitest-driver#1267](https://github.com/appium/appium-xcuitest-driver/pull/1267), [WebDriverAgent#458](https://github.com/appium/WebDriverAgent/pull/458)
+* Fix performance degration of interaction in `nativeWebTap: true` [appium-xcuitest-driver/pull/1265](https://github.com/appium/appium-xcuitest-driver#1265)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:56]: performance degradation

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,3 +1,76 @@
+CHANGES IN VERSION 1.21.0 (FROM 1.20.2)
+===================================
+
+Appium 1.21.0 is a minor release
+
+### General
+
+* Minimum NodeJS version moved to 12
+* Support multiple image matches for [Finding and Interacting with Image Elements](http://appium.io/docs/en/advanced-concepts/image-elements/) [appium-base-driver#449](https://github.com/appium/appium-base-driver/pull/449) (@khaong)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:9]: `(@khaong) -> (thanks to @khaong)`

Comment (mykola-mokhnach) [reply CHANGELOG.md:9]: same below

**Review by mykola-mokhnach — APPROVED** [review]

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,3 +1,76 @@
...
+* Fix to handle special characters by `mobile:type` [appium-adb#567](https://github.com/appium/appium-adb/pull/567)
+* Fix to switch to Web Views with same package name, but different process names [appium-android-driver#687](https://github.com/appium/appium-android-driver/pull/687)
+
+### Android(UIAutomator2)
+
+* Add `mobile:` extensions:
+    * `mobile:doubleClickGesture` for double click guesture [appium-uiautomator2-driver#441](https://github.com/appium/appium-uiautomator2-driver/pull/441), [appium-uiautomator2-server#415](https://github.com/appium/appium-uiautomator2-server/pull/415)
+    * `mobile:installMultipleApks` to install multiple apks via `adb install-multiple` command. Please read [mobile:installMultipleApks](https://github.com/appium/appium-uiautomator2-driver/blob/master/README.md#mobile-installmultipleapks) for more details
+* Fix XPath lookup for queries which have `@*` attribute pattern [appium-uiautomator2-server#417](https://github.com/appium/appium-uiautomator2-server/pull/417)
+* Fix to stop screen recording after a session is closed [appium-uiautomator2-driver#439](https://github.com/appium/appium-uiautomator2-driver/pull/439)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md:26]: The same fix has been done for espresso

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,3 +1,76 @@
...
+
+### Android General
+
+* Disable sending of Google analytics data by instruments [appium-uiautomator2-driver#452](https://github.com/appium/appium-uiautomator2-driver/pull/452), [appium-espresso-driver#666](https://github.com/appium/appium-espresso-driver/pull/666)
+* Improve detection of screen lock status [appium/appium-adb#564](https://github.com/appium/appium-adb/pull/564)
+* Add `speed` poptional arameter to mock location [appium-android-driver#681](https://github.com/appium/appium-android-driver/pull/681) (Thanks to @salabogdan)
+* Fix `autoLaunch` to install applications by `app` and `otherApps` capabilities, but does not launch them as same as iOS [appium-uiautomator2-driver#440](https://github.com/appium/appium-uiautomator2-driver/pull/440), [appium-espresso-driver#654](https://github.com/appium/appium-espresso-driver/pull/654), [appium-android-driver#676](https://github.com/appium/appium-android-driver/pull/676)
+* Fix to handle special characters by `mobile:type` [appium-adb#567](https://github.com/appium/appium-adb/pull/567)
```
Comment (KazuCocoa) [inline CHANGELOG.md:17]: rc4 includes this fix.

**Review by KazuCocoa — COMMENTED** [review]
I think all changes listed in this PR are in rc4. Let me check again later

**Comment by KazuCocoa** [comment]
Ran xcuitest-driver tests on my local with Xcode 12.5 and ios 14.5 sim. No blocking error was there.

**Review by jlipps — APPROVED** [review]

File: CHANGELOG.md [hunk 23]
```diff
@@ -1,3 +1,81 @@
...
+* Add `accessible` attribute field into the XML page source [WebDriverAgent#493](https://github.com/appium/WebDriverAgent/pull/493) (Thanks to @TheDarkestDay)
+* Improve performance of the keyboard dismissal logic [appium-xcuitest-driver#1267](https://github.com/appium/appium-xcuitest-driver/pull/1267), [WebDriverAgent#458](https://github.com/appium/WebDriverAgent/pull/458)
+* Fix performance degradation of interaction in `nativeWebTap:true` [appium-xcuitest-driver/pull/1265](https://github.com/appium/appium-xcuitest-driver#1265)
+
+### Mac2
+
+* Add capabilities:
+    * `appium:webDriverAgentMacUrl` to allow Appium to connect to a running WebDriverAgentMac instance at this URL [appium-mac2-driver#41](https://github.com/appium/appium-mac2-driver/pull/41)
+* Add a possibility to provide key names for `macos:keys` extension [appium-mac2-driver#60](https://github.com/appium/appium-mac2-driver/pull/60)
+* Add `hittable` attribute field into the XML page source [appium-mac2-driver#65](https://github.com/appium/appium-mac2-driver/pull/65)
```
Comment (Jiurong-Yang) [inline CHANGELOG.md:63]: @mykola-mokhnach Can you elaborate on what "hittable" means?

Comment (mykola-mokhnach) [reply CHANGELOG.md:63]: https://developer.apple.com/documentation/xctest/xcuielement/1500561-hittable

**Pull request merged**